### PR TITLE
[cpu][cuda] Extend native_group_norm_backward to handle grad from all outputs

### LINKED ISSUE #180740: [Inductor] functorch_dp_cifar10 accuracy regression in dynamic_inductor_torchbench
### Description

`functorch_dp_cifar10` fails accuracy checks in the `dynamic_inductor_torchbench` periodic CI job (CUDA, A10G):

```
functorch_dp_cifar10    FAIL:     accuracy=fail_accuracy, expected=pass
```

The error is a numerical accuracy regression in batch norm parameters:
```
Accuracy failed for key name layer2.0.bn2.bias
RMSE (res-fp64): 0.00606, (ref-fp64): 0.00145, shape=torch.Size([128])
```

### Details

- Only affects `dynamic_inductor_torchbench` (dynamic shapes + inductor), not static shapes
- Consistent across at least 4 consecutive periodic CI runs (April 17-18)
- Inference pass succeeds; only training accuracy fails
- The model compiles successfully (1 graph, 69 ops, 0 graph breaks)

### Expected CSV entry
`benchmarks/dynamo/ci_expected_accuracy/dynamic_inductor_torchbench_inference.csv`: `functorch_dp_cifar10,pass,0`

### CI Impact

Blocks `periodic-dynamo-benchmarks-test / test (dynamic_inductor_torchbench, 1, 2, linux.g5.4xlarge.nvidia.gpu)`.

cc @ezyang @penguinwu @bobrenjc93 @aditvenk @laithsakka @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by benjaminglass1**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* #186947
* #186946
* #186414
* __->__ #183980

Adds support for gradients from `mean` and `rstd` to `native_group_norm_backward`, via a new overload.  Fixes a broken Inductor benchmark in the pipeline, as well as three XFAILed tests.

Motivation: when running OpInfo tests with TorchInductor, gradient tests are done with all possible gradients supplied.  In the case of `native_group_norm`, this previously invoked `infinitely_differentiable_native_group_norm_backward`, which was written in the style of a `CompositeImplicitAutograd` op.  No explicit handling of input dtypes is done, which results in undesired rounding for float16, causing test failures.

Implementation: a new overload is added for `native_group_norm_backward`, `multiple_grads`, which supports the two additional gradients _and_ missing gradients for the main output (necessary for completeness).  This is a small change to the existing math (two extra subtractions per group), but allows us to test the same way we run.  This allowed me to find and fix the underlying bug from #180740, which was two missing casts from T to T_ACC in `Compute1dBackwardFusedParamsCUDAKernel`.

Note for reviewers: this PR is large by line count, but about half of it is re-indenting existing code or other non-functional changes. In terms of design decisions made, most of the implementation is straightforward and mirrors the existing math from `infinitely_differentiable_native_group_norm_backward`, just at higher precision. The main things I'd like review on are:

1. Does the interface for the new backward overload make sense?
2. The decomposition. The math changes are identical across the different implementations, and well tested (at this point) thanks to `OpInfo` testing, so the decomposition should be a relatively representative thing to review. You'll note that the existing `.out` operator decomposition has been folded into the main one; this allows this decomp to implicitly handle `multiple_grads` without breaking front-end FC.
3. Is it acceptable to add MPS and XPU kernels for this new overload in follow-on PRs?

Fixes #180740

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by benjaminglass1** [comment]
@claude review these changes

File: aten/src/ATen/mps/MPSFallback.mm [hunk 1]
```diff
@@ -114,6 +114,7 @@ static Tensor slow_conv2d_forward_mps(const Tensor& self,
   m.impl("embedding_renorm_", torch::CppFunction::makeFromBoxedFunction<&mps_fallback>());
   m.impl("linalg_svd", torch::CppFunction::makeFromBoxedFunction<&mps_fallback>());
   m.impl("linalg_svd.U", torch::CppFunction::makeFromBoxedFunction<&mps_fallback>());
+  m.impl("native_group_norm_backward.multiple_grads", torch::CppFunction::makeFromBoxedFunction<&mps_fallback>());
```
Comment (benjaminglass1) [inline aten/src/ATen/mps/MPSFallback.mm:115]: I'm planning on implementing this in a future PR to reduce the diff size in this one.

**Comment by benjaminglass1** [comment]
@claude review this code

**Comment by jansel** [comment]
@claude explain those CI failures, are they related?

**Comment by benjaminglass1** [comment]
@jansel I'm tracing down the test failures now; I had an idea on Friday for a missing OpInfo test case that I put before this PR in the stack, and that test case seems to have in fact been broken. I'll push the fix as soon as I have it, but I continue to want review on the work.

**Comment by jansel** [comment]
I'll let others handle this review.

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Review by jansel — CHANGES_REQUESTED** [review]
fix test failures

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Review by jansel — CHANGES_REQUESTED** [review]
fix test failures

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by benjaminglass1** [comment]
Rebased to remove closed PR from the stack, but no code changes were made.

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by benjaminglass1** [comment]
Rebased to solve merge conflicts. I would very much like review on this PR!

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by benjaminglass1** [comment]
Superseded by https://github.com/pytorch/pytorch/pull/186414
